bitcoin/src/blockdata/witness.rs

```diff
@@ -481,9 +485,9 @@ impl<'a> Iterator for Iter<'a> {
 
     fn next(&mut self) -> Option<Self::Item> {
         let index = decode_cursor(self.inner, self.indices_start, self.current_index)?;
-        let varint = VarInt::consensus_decode(&mut &self.inner[index..]).ok()?;
-        let start = index + varint.size();
-        let end = start + varint.0 as usize;
+        let (compact, size) = compact_size::decode(&self.inner[index..]).ok()?;
```

Comment:
I believe we should `.expect("the contents of Witness has always valid compact sizes")` here

Reply:
What about if someone deserializes an invalid witness then tries to iterate over it, no reason for that to panic, right? Or is the solution that we should be doing validation on the `Witness` when deserializing?

Reply:
Raised #3281